### ISSUE TITLE
Issue with macro GENERATED_BODY()

### ISSUE THREAD

**Issue by shadd-anderson** [dotBunny/CLionSourceCodeAccess#56]
I've got the plugin working, however it has an issue with the `GENERATED_BODY()` macro. A bunch of errors get thrown up as shown below:

![image](https://cloud.githubusercontent.com/assets/19513513/26759707/bf5cdc0a-48cb-11e7-874e-8e54ba128670.png)

It doesn't seem to affect anything other than an unsightly red squiggly, but curious nonetheless

**Comment by RicardoEPRodrigues** [comment]
I have this issue often.

I fix it by following these steps:

1. In UE4, press `File > Generate CMakeList`
1. In CLion,  press `File > Reload CMake Project`
1. Wait a bit to reload and it should be fixed.

If that doesn't help, close CLion, repeat the first step and reopen CLion.

It normally fixes the issue.

**Comment by shadd-anderson** [comment]
@RicardoEPRodrigues I tried both solutions and neither of them worked... :/

**Comment by RicardoEPRodrigues** [comment]
Have you installed the UE4 plugin for CLion?

https://plugins.jetbrains.com/plugin/7753-unreal-engine-4-sdk-support

I don't know if it helps, but you could try it out.

**Comment by shadd-anderson** [comment]
Yeah, I've got it installed. Not sure how much it does, tbh. lol

**Comment by shadd-anderson** [comment]
I've noticed it has issues with a few different macros. Might be all of them. `UE_LOG` complains a bunch too

**Comment by RicardoEPRodrigues** [comment]
The code compiles but CLion does not recognize the tags.

Could you tell us your OS and some details. UE4 version for example. Maybe we can find some solution.

**Comment by reapazor** [comment]
Have you allowed CLion to update symbols?

**Comment by shadd-anderson** [comment]
Right. It's just red squigglies. Unreal still compiles no problem.

I'm running Windows 10, UE4 version 4.16.1, CLion version 2017.1.2

I've reloaded CLion multiple times. Remade the CMake file as well.

**Comment by reapazor** [comment]
UE_LOG always seems to complain about inappropriate number of arguments, I've filed a ticket with JetBrains prior about that, but mostly things work out ...

Some things I've noticed:

- if you do not have the build targets setup correctly the project that Unreal generates seems to have a hard time finding references.
- sometimes CLion takes a bit to load symbols, even after load
- latest EAP 2017.2 actually fails on one of our machines to parse anything (bug filed already)

What mine on a good day looks like.

<img width="476" alt="screenshot 2017-06-05 18 35 10" src="https://cloud.githubusercontent.com/assets/99292/26806222/bdbae18c-4a1d-11e7-8e61-6a0d1fa10498.png">

**Comment by shadd-anderson** [comment]
Cool. Thanks @reapazor ! Like I said, I'm not too worried about it, cause it still works. There's no performance issues, it's just a bit of an OCD thing on my side :) Someday soon Unreal and CLion will get along in perfect harmony ;)

**Comment by shadd-anderson** [comment]
Alright, so now I'm having issues that do affect usability, this time with different macros.

I've got an FHitResult that causes issues when I try to get the Actor from it: 

![image](https://user-images.githubusercontent.com/19513513/26905549-98418e9c-4bad-11e7-9bff-d38132cd9d9c.png)

You can see the AActor* pointer isn't recognized, and ActorHit gives an "undefined" error. So I looked into FHitResult, and here's what I found:

![image](https://user-images.githubusercontent.com/19513513/26905584-c9ef580c-4bad-11e7-9032-166d4f08a6e7.png)

Looks like macros in general just aren't working properly. I've tried the suggestions above and it's still not working.

**Comment by shadd-anderson** [comment]
Looks like they've fixed the parameter issue

**Comment by reapazor** [comment]
Indeed !

--
Matthew Davey

On Jul 20, 2017, at 02:49, Shadd Anderson <notifications@github.com<mailto:notifications@github.com>> wrote:


Looks like they've fixed the parameter issue

**Comment by shadd-anderson** [comment]
![image](https://user-images.githubusercontent.com/19513513/28441672-98d5eaf2-6d71-11e7-8852-9e1a97a113be.png)

ahhh... No errors. Feels good
